tools/data-model-exporter/data_model_exporter/dmExporter.py

```diff
@@ -0,0 +1,164 @@
+                try:
+                    properties[prop.n3(g.namespace_manager)] = {
+                        'description': ref,
+                        # TODO: pull the URL reference out of the graph
+                        '$ref': rdfsRangeValue,
+                    }
+                except Exception as e:
+                    raise e
+                    print ("ERROR:"+str(e))
+                    #exit()
+
+                # should limit this to exactly 1 (take the cardinality seriously)
+                if cardinality and cardinality.value == 1:
+                    required.append(prop.n3(g.namespace_manager))
+
+        json_schema = {
+            '$id': rdf_term,
+            '$schema': "http://json-schema.org/draft-07/schema#/",
+            'title': g.value(rdf_term, RDFS.label),
+            # json convention uses a "description" field as the definition
+            'description': str(g.value(rdf_term, Prov.definition)),
+            'definitions': {},
+            'type': 'object',
+            'additionalProperties': True,
+            'properties': properties,
+            'required': required
+        }
+        return json_schema
+
+
+# class to implement RDF to JSON transformation
+# todo: DSPDC-1537
+# Per the spec linked in the parent epic, we will need to implement a method that consumes as input:
+#  A list of class names (this will be parsed from a file later on, but can be hardcoded or passed via the CLI to start for testing purposes)
+#  The path to the relevant TIM TTL file (again can be hardcoded or just threaded through via the CLI)
+#  Execute the processing logic as defined in the spec
+#  A sample spike script for how this would be approached can be found here.
+def rdf_to_json(filePath, classList):
+    #   Edge Cases:
+    #   We should use RDF lib, it can easily parse TTL files and serialize to JSON schema (https://rdflib.readthedocs.io for parsing
+    #   If the path to the TTL file is invalid, RDFLib will barf; we should handle this and output a useful error message
+    #   This should not happen because they should be being created by a tool that creates valid TTL files.
+    #   We will be punting on rdfs:subClassOf properties as there is an open question as to how we'll represent the parent classes in a json schema
+
+    # iterate over classList
+    jsonSchemaList = {}
+    for class_name in classList:
+        # extract json for each individual class
+        jsonSchemaList[class_name] = run(filePath, class_name)
+    return jsonSchemaList
+
+def main():
+    # get CLI arguments
+    filePath, classList = get_arguments()
+    # invoke driver to transform RDF to JSON
+    jsonDict = rdf_to_json(filePath, classList)
+    # write one file per class provided
+    for key in jsonDict:
+        outFileName = "{0}.json".format(key)
+        with open(outFileName, 'w') as f:
+            # TODO: print the json output to the terminal
```

Comment:
Is this still a TODO? It seems like we are in fact printing the json output the terminal below